org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java

```diff
@@ -288,14 +293,53 @@ private CompletionList collectAttributeValueSuggestions(int valueStart, int valu
 		return completionResponse;
 	}
 
-	private CompletionList collectOpenTagSuggestions(int tokenOffset, int tokenEnd) {
-		// TODO Auto-generated method stub
-		return null;
+	//Tags that look like '<{CHARS}'
+	private CompletionList collectOpenTagSuggestions(int tokenOffset, int tokenEnd, String tag) {
+		XMLDocument xmlDocument = completionRequest.getXMLDocument();
+		List<CompletionItem> list= new ArrayList<CompletionItem>();
+		if(xmlDocument.children.size() > 0){
+			Position start,end;
+			try {
+				start = xmlDocument.positionAt(tokenOffset);
+				end = xmlDocument.positionAt(tokenEnd);	
+				
+			} catch (Exception e) {
+				return new CompletionList();
+			}
+			Node node = xmlDocument;
+			Range range = new Range(start,end);
+			if(cdata.regionMatches(0, tag, 0, tag.length())){
+				//TODO: setCDATACompletionItem(list,range);
+			}
+			else{
+				collectSimilarTags(list, node, tag, range);
+			}
+			CompletionList p = new CompletionList(list);
+			return p;
+		}
+		return new CompletionList();
 	}
 
+	//Tags that look like '<'
 	private CompletionList collectTagSuggestions(int offset, int endPos) {
-		// TODO Auto-generated method stub
-		return null;
+		XMLDocument xmlDocument = completionRequest.getXMLDocument();
+		List<CompletionItem> list= new ArrayList<CompletionItem>();

```

Comment:
declare that list right before using it (L336), else it might be useless resource allocation